tests/main/completion/lib.exp

```diff
@@ -0,0 +1,66 @@
+set send_slow {1 0.1}
+set timeout 5
+
+proc delay {} {
+    sleep 0.1
+}
+
+proc chat {outstr instr {keep false}} {
+    send $outstr
+    delay
+    if $keep {
+        expect {
+            timeout {puts "timeout"; exit 1}
+            -notransfer $instr
```

Comment:
expect moves the string it's looking at forward to just after the last thing it found, so if it has "hello world" and you expect hello, the next expect will have " world". `-notransfer` changes the behaviour to retain the whole "hello world" (plus anything new that might come in). It's called transfer because it has a series of buffers, one is the one it's looking at, at another is the "out buffer", and expect usually moves stuff that matched to the out buffer.